cluster-controller/src/main/java/io/strimzi/controller/cluster/resources/Labels.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.controller.cluster.resources;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonMap;
+import static java.util.Collections.unmodifiableMap;
+
+/**
+ * An immutable set of labels
+ */
+public class Labels {
+
+    public static final String STRIMZI_DOMAIN = "strimzi.io/";
+
+    /**
+     * The kind of a ConfigMap:
+     * <ul>
+     *     <li>{@code strimzi.io/kind=cluster}
+     *         identifies a ConfigMap that is intended to be consumed by
+     *         the cluster controller.</li>
+     *     <li>{@code strimzi.io/kind=topic}
+     *         identifies a ConfigMap that is intended to be consumed
+     *         by the topic controller.</li>
+     * </ul>
+     */
+    public static final String STRIMZI_KIND_LABEL = STRIMZI_DOMAIN + "kind";
+    /**
+     * The type of Strimzi component:
+     * E.g: {@code kafka}, {@code zookeeper}, {@code topic-controller},
+     *      {@code connect}, {@code connect-s2i}
+     */
+    public static final String STRIMZI_TYPE_LABEL = STRIMZI_DOMAIN + "type";
+
+    /**
+     * The Strimzi cluster the resource is part of.
+     * The value is the cluster name (i.e. the name of the cluster CM)
+     */
+    public static final String STRIMZI_CLUSTER_LABEL = STRIMZI_DOMAIN + "cluster";
+
+    /**
+     * The name of the K8S resource.
+     * This is often the same name as the cluster
+     * (i.e. the same as {@code strimzi.io/cluster})
+     * but is different in some cases (e.g. headful and headless services)
+     */
+    public static final String STRIMZI_NAME_LABEL = STRIMZI_DOMAIN + "name";
+
+    /**
+     * The empty set of labels.
+     */
+    public static final Labels EMPTY = new Labels(emptyMap());
+
+    private final Map<String, String> labels;
+
+    /**
+     * Returns the value of the {@code strimzi.io/cluster} label of the given {@code resource}.
+     */
+    public static String clusterLabel(HasMetadata resource) {
```

Comment:
Just thinking ... Do we really need these methods to be called `clusterLabel`? Wouldn't `cluster` be enough?

Reply:
There's already a `public static Labels cluster(String)` method which "goes the other way", returning a singleton `Labels` instance with just the `strimzi.io/cluster` label. It would be confusing to have overloaded methods which did completely different things. But maybe that one should be called `clusterSingleton()` or `forCluster()`. Wdyt?